build_exercise_proj/CMakeLists.txt

```
@@ -1,24 +1,25 @@
-cmake_minimum_required(VERSION 3.1)
-project(build_exercise)
+cmake_minimum_required(VERSION 3.1) 
+project(build_exercise) 
 
-set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON) 
 
-set(_INC_FOLDER_NAME include)
-set(_INC_DIR ${CMAKE_CURRENT_SOURCE_DIR}/${_INC_FOLDER_NAME})
-set(_SRC_DIR ${CMAKE_CURRENT_SOURCE_DIR}/src)
+set(_INC_FOLDER_NAME include) 
+set(_INC_DIR ${CMAKE_CURRENT_SOURCE_DIR}/${_INC_FOLDER_NAME}) 
+set(_SRC_DIR ${CMAKE_CURRENT_SOURCE_DIR}/src) 
 
 
-add_executable(
-    ${PROJECT_NAME}     
-        ${_SRC_DIR}/main.cpp
+add_executable( 
+    ${PROJECT_NAME}      
+        ${_SRC_DIR}/main.cpp 
 )
 
-add_subdirectory(tools_lib_proj)
+add_subdirectory(tools_lib_proj) 
 
-target_include_directories(${PROJECT_NAME} PUBLIC tools_lib_proj)
+target_include_directories(${PROJECT_NAME} PUBLIC tools_lib_proj) 
 
-target_link_directories(${PROJECT_NAME} PRIVATE tools_lib_proj)
+target_link_directories(${PROJECT_NAME} PRIVATE tools_lib_proj) 
```

The method is
```suggestion
target_link_libraries(${PROJECT_NAME} PRIVATE tools_lib_proj) 
```
Not "target_link_directories"

A call to `target_link_directories` will automatically bring in the includes exported from "tools_lib_proj".

So you don't need to do `target_include_directories(${PROJECT_NAME} PUBLIC tools_lib_proj)`.
Instead, you should do 
```
target_include_directories(${PROJECT_NAME} PUBLIC ${_INC_DIR})
```
INSIDE the "tools_lib_proj" CMakeLists.txt.
This way every consumer of this package (in this case the project "build_exersice" will translatively get those includes when it uses `target_link_libraries`

---

In the code snippet did you mean to put "target_link_libraries" ?